doc/profiling/sonic_bootchart.md

```diff
@@ -0,0 +1,202 @@
+Running command: systemctl enable systemd-bootchart
+admin@sonic:~$ sudo sonic-bootchart disable
+Running command: systemctl disable systemd-bootchart
+```
+
+In case image is built without bootchart included (INCLUDE_BOOTCHART=n) sonic-bootchart will print an error:
+
+```
+admin@sonic:~$ sudo sonic-bootchart enable
+systemd-bootchart is not installed
+```
+
+Once bootchart is enabled a systemd-bootchart.service is enabled in systemd which starts a daemon early at boot that collects samples.
+This setting is permanent, meaning, after it is enabled, bootchart will always run at boot until disabled, *config save*, *config reload* do not affect this setting.
+
+
+Update configuration example:
+
+```
+admin@sonic:~$ sudo sonic-bootchart config --samples 500 --frequency 10
+```
+
+This command will update /etc/systemd/bootchart.conf. The next time bootchart will run (at system start after reboot) it will take the new values.
+*systemd-bootchart* saves the plots to a temporary directory /run/log that is flushed after reboot.
+
+Displaying configuration, operational status and output SVG plots:
+
+```
+admin@sonic:~$ sudo sonic-bootchart show
+Status      Operational Status    Samples    Frequency    Time (sec)  Output
+--------  --------------------  ---------  -----------  ------------  ------------------------------------
+enabled              in-active        500           10            50  /run/log/bootchart-20220504-1325.svg
+```
+
+Fields description:
+
+| Field |  Comment |
+|-------|--------------------------|
+| Status | Output of "systemctl is-enabled systemd-bootchart". Shows whether this service is configured to start at boot (enabled/disabled) |
+| Operational Status | Output of "systemctl is-action systemd-bootchart". Shows whether this service is currently collecting samples (active/in-active) |
+| Samples | Configured amount of samples to collect |
+| Frequency | How frequent to collect samples per second |
+| Time | The time how long after boot samples will be collected, it is a result of division samples/frequency |
+| Output | If systemd-bootchart finished collecting samples (in-active) this column will display resulting plots, otherwise this column is empty |
+
+
+Usage example:
+
+- First make sure you have SONiC image built with INCLUDE_BOOTCHART=y
+- Enable bootchart: "sudo sonic-bootchart enable"
+- Perform any kind of reboot: "sudo reboot" or "sudo warm-reboot" or "sudo fast-reboot"
+- Wait till system reboots and the plot is generated by quering "sudo sonic-bootchart show"
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A
+
+#### CLI/YANG model Enhancements
+
+N/A
+
+#### Config DB Enhancements
+
+N/A
+
+### Warmboot and Fastboot Design Impact
+N/A
+
+### Restrictions/Limitations
+
+### Testing Requirements/Design
+N/A
+
+#### Unit Test cases
+
+Cover CLI with UT for:
+  - enabling bootchart
+  - disabling bootchart
+  - config/show commands
+
+#### System Test cases
+
+N/A
+
+### Open/Action items
```

Comment:
Suggest adding code at initramfs stage to generate bootchart configuration file according to kernel parameters. So that we could have bootchart configured for upcoming boot generated.

Reply:
Instead of initramfs we may add this to sonic-installer (where we usually perform config/packages migration tasks).
Added an open action.